test/jdk/tools/jpackage/helpers/jdk/jpackage/test/JPackageCommand.java

```diff
@@ -967,7 +967,12 @@ public static enum ReadOnlyPathAssert{
         APP_CONTENT(new Builder("--app-content").multiple().create()),
         RESOURCE_DIR(new Builder("--resource-dir").create()),
         MAC_DMG_CONTENT(new Builder("--mac-dmg-content").multiple().create()),
-        RUNTIME_IMAGE(new Builder("--runtime-image").create());
+        RUNTIME_IMAGE(new Builder("--runtime-image").enable(cmd -> {
+            // External runtime image should be R/O unless it is runtime installer
+            // on macOS. On macOS runtime image will be signed ad-hoc or with
+            // real certificate when creating runtime installers.
+            return !(cmd.isRuntime() && TKit.isOSX());
```

Comment:
I guess, the directory referenced from the `--runtime-image` option will not be signed if it is a JDK image, not a JDK bundle. So this switch turns off verification in cases when the verification should be done.